### ISSUE TITLE
  Cinnamon Maximus doesnt work properly on cinnamon 3.4 new windows when maximized disappear until cinnamon is restarted.

### ISSUE THREAD

**Issue by mikee805** [Odyseus/CinnamonTools#74]
### Description

Cinnamon Maximus doesnt work properly on cinnamon 3.4 new windows when maximized disappear until cinnamon is restarted.

### Steps to reproduce

1. open a new application
2. maximize it

OR

1. open a maximized application
2. it has no visible window

**Expected behavior:**
Window decorators disappear but window stays

**Actual behavior:**
Window disappears

**Reproduces how often:** 
happens everytime

### System information

inxi -xxxSc0 -! 31
System:    Kernel: 4.8.0-51-generic x86_64 (64 bit gcc: 5.4.0)
           Desktop: Cinnamon 3.4.0 (Gtk 3.18.9-1ubuntu3.3) dm: lightdm
           Distro: Ubuntu 16.04 xenial

### System logs

nothing relevant in ~/.cinnamon/glass.log
from ~/.xsession-errors not sure it relevant:

```Cjs-Message: JS LOG: Invalid or null source id used when attempting to run Mainloop.source_remove()
== Stack trace for context 0x55c744ace030 ==
overrideMainloop/Mainloop.source_remove@/usr/share/cinnamon/js/ui/overrides.js:114:13
MyApplet.prototype._onLEntered@/home/mepstein/.local/share/cinnamon/applets/SW++@mohammad-sn/applet.js:446:13
PopupBaseMenuItem.prototype.setActive@/usr/share/cinnamon/js/ui/popupMenu.js:186:51
PopupBaseMenuItem.prototype._onHoverChanged@/usr/share/cinnamon/js/ui/popupMenu.js:173:9

Cjs-Message: JS LOG: Invalid or null source id used when attempting to run Mainloop.source_remove()
== Stack trace for context 0x55c744ace030 ==
overrideMainloop/Mainloop.source_remove@/usr/share/cinnamon/js/ui/overrides.js:114:13
MyApplet.prototype._onLEntered@/home/mepstein/.local/share/cinnamon/applets/SW++@mohammad-sn/applet.js:446:13
PopupBaseMenuItem.prototype.setActive@/usr/share/cinnamon/js/ui/popupMenu.js:186:51
PopupBaseMenuItem.prototype._onHoverChanged@/usr/share/cinnamon/js/ui/popupMenu.js:173:9

Cjs-Message: JS LOG: Invalid or null source id used when attempting to run Mainloop.source_remove()
== Stack trace for context 0x55c744ace030 ==
overrideMainloop/Mainloop.source_remove@/usr/share/cinnamon/js/ui/overrides.js:114:13
MyApplet.prototype._onLEntered@/home/mepstein/.local/share/cinnamon/applets/SW++@mohammad-sn/applet.js:446:13
popModal@/usr/share/cinnamon/js/ui/main.js:1343:9
PopupMenuManager.prototype._ungrab@/usr/share/cinnamon/js/ui/popupMenu.js:3435:9
PopupMenuManager.prototype._onMenuOpenState@/usr/share/cinnamon/js/ui/popupMenu.js:3471:17
_emit@resource:///org/cinnamon/cjs/modules/signals.js:142:27
PopupMenu.prototype.close@/usr/share/cinnamon/js/ui/popupMenu.js:2404:9
PopupMenuBase.prototype._connectItemSignals/menuItem._activateId<@/usr/share/cinnamon/js/ui/popupMenu.js:1861:17
_emit@resource:///org/cinnamon/cjs/modules/signals.js:142:27
PopupBaseMenuItem.prototype.activate@/usr/share/cinnamon/js/ui/popupMenu.js:177:9
PopupBaseMenuItem.prototype._onButtonReleaseEvent@/usr/share/cinnamon/js/ui/popupMenu.js:150:9

Cjs-Message: JS LOG: Invalid or null source id used when attempting to run Mainloop.source_remove()
== Stack trace for context 0x55c744ace030 ==
overrideMainloop/Mainloop.source_remove@/usr/share/cinnamon/js/ui/overrides.js:114:13
MyApplet.prototype._onLEntered@/home/mepstein/.local/share/cinnamon/applets/SW++@mohammad-sn/applet.js:446:13

Cinnamon WARNING: Possible orphan label being accessed via st_label_set_text().  Check your timers and handlers!
Address: 0x55c74622cb90  Text: 0
```
Thanks for the great plugin was perfect until 3.4 :)

**Comment by Odyseus** [comment]
Hello, @mikee805.

See if any of what I said [here](https://github.com/Odyseus/CinnamonTools/issues/55#issuecomment-299619433) helps. It refers to a tweak in the **Cinnamon Tweaks** extension, but it applies to the Maximus extension as well because its the exact same code. See if the alternative option suits your needs too.

**Comment by mikee805** [comment]
@Odyseus thanks for the link but I can get that to work :(
from: /usr/share/themes/Adwaita/metacity-1/metacity-theme-3.xml:
```<frame_geometry name="max" has_title="false" title_scale="medium" parent="normal" rounded_top_left="false" rounded_top_right="false">
	<distance name="left_width" value="0" />
	<distance name="right_width" value="0" />
	<distance name="title_vertical_pad" value="0"/>
	<distance name="right_titlebar_edge" value="0"/>
	<distance name="left_titlebar_edge" value="0"/>
	<border name="title_border" left="0" right="0" top="0" bottom="0"/>
	<border name="button_border" left="0" right="0" top="0" bottom="0"/>
	<distance name="bottom_height" value="0" />
    <distance name="button_width" value="0"/>
	<distance name="buttom_height" value="0" />
</frame_geometry>
```
even rebooted and the max windows still have decorators

**Comment by Odyseus** [comment]
@mikee805.

If you used the exact same code that you attached to your message, then it will never work. For the Adwaita theme that comes with Linux Mint, the following works.

```xml
<frame_geometry name="max" has_title="false" title_scale="medium" parent="normal" rounded_top_left="false" rounded_top_right="false">
    <distance name="right_width" value="0" />
    <distance name="left_titlebar_edge" value="0"/>
    <distance name="right_titlebar_edge" value="0"/>
    <distance name="title_vertical_pad" value="0"/>
    <border name="title_border" left="0" right="0" top="0" bottom="0"/>
    <border name="button_border" left="0" right="0" top="0" bottom="0"/>
</frame_geometry>
```

**Comment by mikee805** [comment]
@Odyseus wow your amazing thanks so much!

**Comment by Odyseus** [comment]
Hello, @mikee805.

I tracked down the problem of invisible windows and fixed it (at least, I cannot reproduce it anymore on Cinnamon 3.4.x). If you are still using the extension, with the next update (I published it just now and awaiting review) there will be a new option called **Enable hack for invisible windows**. After enabling this option, all windows with decorations removed needs to be restarted. Read the option tooltip for know *side effects*.
